squad/core/models.py

```diff
@@ -1058,20 +1058,31 @@ def get_fixes(self):
     def get_exceeded_thresholds(self):
         # Return a list of all (threshold, metric) objects for those
         # thresholds that were exceeded by corresponding metrics.
+        environments_qs = self.build.project.environments.all()
         thresholds_exceeded = []
         if self.has_metrics:
             test_runs = self.build.test_runs.all()
             suites = Suite.objects.filter(test__test_run__build=self.build)
+            thresholds = MetricThreshold.objects.filter(
+                environment__in=environments_qs.values_list('id', flat=True)
```

Comment:
You're making two queries here:
1. querying all environments for this project
2. gathering all ids and passing it as `environment_id IN (...)` in the metricThreshold query

You can make it a single, an faster, query by doing:

```python
thresholds = MetricThreshold.objects.filter(environment__in=self.build.project.environments.all())
```